### PR TITLE
FEAT : Add Model Dashboard

### LINKED ISSUE #221: ENH: gradio for pre-trained models
### Is your feature request related to a problem? Please describe
It is nice to have a gradio page for generate models, similar to:
image

### PR DESCRIPTION
**Pull request by Bojun-Feng**
Replace the default Gradio page with a React.js framework for a multi-page dashboard. Right now there is only one page named "Running Models", which displays all running models, launches Gradio WebUIs for them, and terminates them. The file structure is set up such that it should be easy to add more pages in the future.

Created as a first step in resolving #93. Resolves #221.

Here is the file structure overview:
- Added folder `/inference/xinference/web/ui/`, where all frontend code is hosted
- Each shared or public component used by multiple pages is stored `ui/src/components` folder
- Each page has its own subdirectory in the `ui/src/scenes` folder
- The `ui/src/App.js` file is used to route the pages and link them together

To-do list:
- [x] edit or remove README.md in `/inference/xinference/web/ui/`
- [x] test ChatGLM compactibility

### CODE REVIEW

File: xinference/core/restful_api.py [hunk 1]
```diff
@@ -309,6 +357,7 @@ async def launch_model(self, request: Request) -> JSONResponse:
         quantization = payload.get("quantization")
 
         exclude_keys = {
+            "endpoint",
```
Comment (UranusSeven) [inline xinference/core/restful_api.py:360]: Is this still needed?

File: xinference/core/restful_api.py [hunk 2]
```diff
@@ -266,11 +274,51 @@ def serve(self):
             methods=["POST"],
             response_model=ChatCompletion,
         )
-        app.include_router(self._router)
-        app = gr.mount_gradio_app(app, self._gradio_block, path="/")
+
+        self._router.add_api_route(
+            "/v1/gradio/{model_uid}", self.build_interface, methods=["POST"]
```
Comment (UranusSeven) [inline xinference/core/restful_api.py:279]: You might prefer to design the API in a way that's unrelated to the library we rely on.

How about `/v1/ui/model_uid`?

File: xinference/core/restful_api.py [hunk 3]
```diff
@@ -266,11 +274,51 @@ def serve(self):
...
+
+            @self._app.get("/")
+            def read_main():
+                response = RedirectResponse(url="/ui/")
+                return response
+
+            self._app.mount(
+                "/ui/",
+                SPAStaticFiles(directory=ui_location, html=True),
+                name="Xinference",
```
Comment (UranusSeven) [inline xinference/core/restful_api.py:309]: What does `name` do?

File: xinference/web/ui/src/media/icon.png
Comment (UranusSeven) [inline xinference/web/ui/src/media/icon.png:1]: Is this still needed?

File: xinference/web/ui/src/media/icon.webp
Comment (UranusSeven) [inline xinference/web/ui/src/media/icon.webp:1]: Is this still needed?

File: xinference/web/ui/src/media/logo_vicuna.webp
Comment (UranusSeven) [inline xinference/web/ui/src/media/logo_vicuna.webp:1]: Is this still needed?

File: xinference/deploy/supervisor.py [hunk 4]
```diff
@@ -60,6 +60,7 @@ async def start_supervisor_components(address: str, host: str, port: int):
         uid=RESTfulAPIActor.uid(),
         sockets=sockets,
         gradio_block=gradio_block,
+        endpoint=f"http://{host}:{port}",
```
Comment (UranusSeven) [inline xinference/deploy/supervisor.py:63]: Does it work when host is '0.0.0.0'?

File: xinference/core/restful_api.py (hunk 3 above)
Comment (Bojun-Feng) [reply xinference/core/restful_api.py]: I believe `name` is supposed to give the webpage a name or header. In this case, it did not work due to the redirection from the main page. I have removed this line.

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**